### DISCUSSION TITLE
Widgets Menu gets disabled by "Taskbar Style: Windows 10"

### DISCUSSION THREAD

**Discussion by ghost** [valinet/ExplorerPatcher#1465] [Q&A]
The widgets menu that appears when pressing Left Windows + W will not appear when Windows 11 starts up with the ExplorerPatcher set to "Taskbar Style: Windows 10".

It IS possible to get the menu to work again by going into ExplorerPatcher, changing the setting to "Taskbar Style: Windows 11 (default)", changing the setting back to "Taskbar Style: Windows 10", and restarting File Explorer.

I ended up creating a series of macros and scheduled tasks to switch the setting, restart file explorer, and then refresh some apps that get glitched tray icons, but this is not exactly ideal.

Is there a way to change the setting back and forth with a command line that I could put into a batch file or PowerShell script to run during startup?

**Comment by SD308** [comment]
Or perhaps have widgets as an application view thing? Like many other apps. Or even similar to the weather widget that is available.

**Comment by pyrates999** [comment]
Explorer patcher stores its settings in the registry key: "Computer\HKEY_CURRENT_USER\Software\ExplorerPatcher"

If you can figure out what is changed, you could do it from the command line.  You'd have to restart file explorer each time though.

**Comment by SD308** [comment]
Wind 11 Taskbar                                   Wind 10
NA                                                        NA
0                                                           0
1                                                           1
0                                                           0
0                                                           0
1                                                           1
0                                                           0
0                                                           1
0                                                           0
10.0.22621.1848                                  10.0.22621.1848    
0                                                           0
0                                                           0
1                                                           1
1                                                           1
0                                                           0
1                                                           1
3                                                           3
0                                                           0
sws NA                                                 NA
twinui                                                        
NA                                                        NA
2469e0                                                 2469e0          
65c044                                                 65c044          
65c37c                                                 65c37c          
64a050                                                64a050          
64b6a0                                                64b6a0           
69db3c                                                69db3c           
69e344                                                69e344          
2d2d0                                                  2d2d0         
4d461c                                                 4d461c          


Seems that the only change is registry OldTaskbar. This seems like it wouldn't allow the widgets AND the windows 10 look.

**Reply by ghost** [reply]
I'm not sure, because after a fresh login to W11, if I:

1. change the taskbar setting from 10 to 11
2. restart file explorer

the widgets menu will still not appear with LWIN+W.

However, if, after a fresh login I:

1. try to load widgets menu with LWIN+W (which fails)
2. change the taskbar setting from 10 to 11
3. restart file explorer

the widgets menu will pop out while file explorer is restarting, and then it will appear any time I press LWIN+W until the next logoff or restart. It seems that Windows either does load the menu, but it is hidden somehow offscreen, or the menu animation is queued up and only fires if file explorer gets restarted.

It is also interesting to note that I used to be able to press the LWIN key , which would bring up the start menu, and then I could immediately start typing any letters to automatically bring up the search menu.

Nowadays that does not bring up the menu, but if I do start typing then the "active window" focus in the OS seems to get stolen away from the start menu such that pressing the ESC key will not close the start menu, until I click somewhere on the start menu and then press ESC. This leads me to believe these elements are actually working, but just hidden somewhere offscreen or behind some layer.

As for changing the OldTaskbar registry settings, I could probably figure out how to create such a batch file or powershell script to do that and then restart explorer.exe. Maybe even an AutoHotKey script.

**Reply by pyrates999** [reply]
Widgets weren't part of windows 10.  So that's why they may not be appearing when you switch to the windows 10 taskbar.  Their code may only be in the windows 11 taskbar.

**Comment by ghost** [comment]
I ended up making a batch file that does everything:

```
REG ADD HKCU\Software\ExplorerPatcher /v OldTaskbar /t REG_DWORD /d 0 /f
taskkill /im explorer.exe /f
start explorer.exe
REG ADD HKCU\Software\ExplorerPatcher /v OldTaskbar /t REG_DWORD /d 1 /f
taskkill /im explorer.exe /f
start explorer.exe
C:\Windows\System32\schtasks.exe /run /tn "RefreshBadTrayApps"
```

The last line runs a scheduled task "RefreshBadTrayApps" that runs with admin privileges, with the following actions:

Start a Program: TASKKILL; Arguments: /F /IM NVDisplay.Container.exe
Start a Program: TASKKILL; Arguments: /F /IM icue.exe
Start a Program: "C:\Program Files\Corsair\Corsair iCUE5 Software\iCUE Launcher.exe"

The reason for this is because killing and restarting file explorer will cause some apps to behave incorrectly in the system tray. In my case, it's iCUE and NVIDIA Settings. NVDisplay.Container.exe will automatically restart when it is terminated, so only iCUE needs an additional action to run its launcher.

Now the only problem is that I can't run this script automatically at login using Scheduled Tasks, or use another script to run it, because it triggers the crash protection popup from ExplorerPatcher. But I can run it using a macro in iCUE, so I guess that's the best that I will get.
